scipy/optimize/_numdiff.py

```diff
@@ -702,9 +847,33 @@ def _sparse_difference(fun, x0, f0, h, use_one_sided,
     row_indices = np.hstack(row_indices)
     col_indices = np.hstack(col_indices)
     fractions = np.hstack(fractions)
+
     if isspmatrix(structure):
-        return csr_matrix((fractions, (row_indices, col_indices)), shape=(m, n))
-    return csr_array((fractions, (row_indices, col_indices)), shape=(m, n))
+        return csr_matrix((fractions, (row_indices, col_indices)), shape=(m, n)), nfev
+    return csr_array((fractions, (row_indices, col_indices)), shape=(m, n)), nfev
+
+
+class _Fun_Wrapper:
```

Comment:
This is a specific wrapper for  functions used in `approx_derivative`. This function can return an array of different dimensions, unlike `_ScalarFunctionWrapper` which is designed to only return a scalar.